infoboxes/commons/infobox.lua

```diff
@@ -0,0 +1,263 @@
+
+        previousWrapper :node(previousArrow)
+                        :wikitext('&nbsp;[[' .. previous .. ']]')
+
+        node:node(previousWrapper)
+    end
+
+    if doesNextExist then
+        local nextWrapper = mw.html.create('div')
+        nextWrapper :addClass('infobox-cell-2')
+                    :addClass('infobox-text-right')
+
+        local nextArrow = mw.html.create('div')
+        nextArrow       :addClass('infobox-arrow-icon')
+                        :css('float', 'right')
+                        :wikitext('[[File:Arrow sans right.svg|link=' .. next .. ']]')
+
+        nextWrapper :wikitext('[[' .. next .. ']]&nbsp;')
+                    :node(nextArrow)
+
+        node:node(nextWrapper)
+    end
+
+    self.chronology:node(node)
+    return self
+end
+
+function Infobox:_createInfoboxButtons()
+    local rootFrame
+    local currentFrame = mw.getCurrentFrame()
+    while currentFrame ~= nil do
+        rootFrame = currentFrame
+        currentFrame = currentFrame:getParent()
+    end
+
+    local moduleTitle = rootFrame:getTitle()
+
+    local buttons = mw.html.create('span')
+    buttons:addClass('infobox-buttons')
+    buttons:node(
+        mw.text.nowiki('[') .. '[' .. mw.site.server ..
+        tostring(mw.uri.localUrl( mw.title.getCurrentTitle().prefixedText, 'action=edit&section=0' )) ..
+        ' e]' .. mw.text.nowiki(']')
+    )
+    buttons:node(
+        mw.text.nowiki('[') ..
+        '[[' .. moduleTitle ..
+        '/doc|h]]' .. mw.text.nowiki(']')
+    )
+
+    return buttons
+end
+
+function Infobox:links(links)
+    local infoboxLinks = mw.html.create('div')
+    infoboxLinks    :addClass('infobox-center')
+                    :addClass('infobox-icons')
+
+    for key, value in pairs(links) do
+		local iconString = Infobox.getIconString(key)
+        local link = '[' .. Links.makeFullLink(key, value) .. ' <i class="lp-icon lp-' .. iconString .. '></i>]'
+        infoboxLinks:wikitext(' ' .. link)
+    end
+
+    self.content:node(mw.html.create('div'):node(infoboxLinks))
+    return self
+end
+
+function Infobox.getIconString(key)
+	return string.gsub(key, '%d$', '')
+end
+
+function Infobox:centeredCell(content)
+    if content == nil then
+        return self
+    end
+
+    local infoboxCenteredCell = mw.html.create('div'):addClass('infobox-center')
+    infoboxCenteredCell:wikitext(content)
+    self.content:node(mw.html.create('div'):node(infoboxCenteredCell))
+    return self
+end
+
+function Infobox:customCell(contents)
+    if type(contents) ~= 'table' or contents == {} then
+        return self
+    end
+
+    local div = mw.html.create('div')
+    local number = #contents
+    for _, content in ipairs(contents) do
+        local infoboxCustomCell = mw.html.create('div'):addClass('infobox-cell-' .. number .. ' infobox-center')
+        infoboxCustomCell:wikitext(content)
+        div:node(infoboxCustomCell)
+    end
+
+    self.content:node(div)
+    return self
+end
```

Comment:
This function should be left out of this PR, `fcell` provides this functionality already

Reply:
see #171